File: server/models/subject_section_history.go
package models

import (
	surrealModels "github.com/surrealdb/surrealdb.go/pkg/models"
)

type SubjectSectionHistory struct {
	ID surrealModels.RecordID `json:"id" validate:"required"  swaggertype:"object"`

	Schedules      []surrealModels.RecordID `json:"schedules" swaggertype:"array,object"`
	SubjectSection surrealModels.RecordID   `json:"subject_section" swaggertype:"object"`
	UserId         surrealModels.RecordID   `json:"user_id" swaggertype:"object"`

	StartDate surrealModels.CustomDateTime  `json:"start_date"`
	EndDate   *surrealModels.CustomDateTime `json:"end_date,omitempty"`

	NewData SubjectSectionEntity `json:"new_data,omitempty"`
}

type SubjectSectionHistoryWithSchedules struct {
	SubjectSectionHistory
	Schedules []SubjectScheduleEntity `json:"schedules"`
}

File: client/src/interfaces/SubjectSectionHistory.ts
import type { Id } from './surrealDb';
import type { SubjectSection } from './SubjectSection';
import type { SubjectSchedule } from './SubjectSchedule';

export interface SubjectSectionHistory {
  id: Id<'subject_section_history'>;

  schedules: Id<'subject_section'>[];
  subject_section: Id<'subject_section'>;
  user_id: Id<'user'>;

  start_date: string;
  end_date: string | null;

  new_data: SubjectSection;
}

export interface SubjectSectionHistoryWithSchedules extends Omit<SubjectSectionHistory, 'schedules'> {
  schedules: SubjectSchedule[];
}
